### PR TITLE
feat: HA for schema registry

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

high availability for schema registry. 
RisingWave will request all provided URLs and return the first successful one. 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

for schema registry related, we gonna allow multiple URLs chained up by `,` like kafka brokers

```
CREATE TABLE upsert_student_avro_json ()
WITH (
	connector = 'kafka',
  properties.bootstrap.server = 'message_queue:29092',
	topic = 'upsert_student_avro_json')
FORMAT UPSERT ENCODE AVRO (schema.registry = 'http://message_queue1:8081,http://message_queue2:8081,http://message_queue3:8081');
```

</details>

### CODE REVIEW

**Comment by tabVersion** [comment]
> Are we going to support this feature in frontend in this pr?

No, we will let user to specify multiple addrs in schema registry

eg. `schema.registry = 'http://127.0.0.1:6789,http://127.0.0.1:7890'`

**Review by wugouzi — APPROVED** [review]
LGTM

**Review by hzxa21 — APPROVED** [review]
Rest LGTM

File: src/connector/src/parser/schema_registry/client.rs [hunk 1]
```diff
@@ -46,36 +61,57 @@ impl Client {
...
+                method.clone(),
+            )));
+        }
 
-        if self.username.is_some() {
-            request = request.basic_auth(self.username.clone().unwrap(), self.password.clone())
+        while !fut_req.is_empty() {
+            let (result, _index, remaining) = select_all(fut_req).await;
+            match result {
+                Ok(Ok(res)) => return Ok(res),
```
[the remark this answers is not in the document]
Comment (tabVersion) [reply src/connector/src/parser/schema_registry/client.rs]: you are right, we can call `abort` in each future.

**Pull request merged**